packages/admin/cms-admin/src/common/routeMenu/routeMenu.type.ts

```diff
@@ -0,0 +1,11 @@
+import { MenuItemRouterLinkProps } from "@comet/admin";
+import { RouteProps } from "react-router";
+
+export type RouteMenuItem = Omit<MenuItemRouterLinkProps, "to"> & {
+    route?: RouteProps;
+    to?: string;
+};
+export type RouteMenuItemWithSubMenu<Depth extends number = 1> = RouteMenuItem & { subMenu?: RouteMenu<Depth> };
+
+type Decr = [never, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+export type RouteMenu<Depth extends number> = Depth extends 1 ? RouteMenuItem[] : RouteMenuItemWithSubMenu<Decr[Depth]>[];
```

Comment:
what do you win with this fancy type? (instead of using a unlimited depth)

(oather that you can show your ts skills :)

Reply:
The only thing is that you can restrict the depth of the Route-Menu data: https://github.com/vivid-planet/comet/pull/1262/files#diff-d560eb6a8e30a118f1c22a036aa46ceafbcd8765ac75055a3d57aeced007a0fdR42

And it was more a googleing skill than a ts skill... ;-)

Reply:
ah, ok that makes sense :+1: